src/renderer/components/NewProjectModal/NewProjectView.tsx

```diff
@@ -53,18 +53,35 @@ const NewProjectView = ({ closeModal, nextView }: Props) => {
 
   const dispatch = useDispatch();
 
-  const setProjectInStore = async (project: Project) => {
-    dispatch(projectCreated(project));
-  };
+  const setProjectInStore = useCallback(
+    async (project: Project) => {
+      dispatch(projectCreated(project));
+    },
+    [dispatch]
+  );
 
-  const handleContinue = async () => {
+  const handleContinue = useCallback(async () => {
     const project = await makeProjectWithoutMedia(projectName);
     if (project === null) {
       return;
     }
     setProjectInStore(project);
     nextView();
-  };
+  }, [nextView, projectName, setProjectInStore]);
+
+  useEffect(() => {
```

Comment:
While I love the hook, wondering if it is possible to just use a form around the text field and button? Then add an `onSubmit` handler to the form and make the button `type="submit"`.

Reply:
I like that approach, I gave it a shot. I did not like the outcome though. I published a temporary branch to show you what I have tried and so you can run it locally and possibly see what is going on.

Here is the commit where I replaced the hook based approach with the form based approach. https://github.com/patrickbrett/mlvet/commit/9962fce5bf0dec5a3c0195f2e5112101f503a67e

The branch name is temporary/naviagte-with-return (yes there is a typo)

Here is what I noticed:
- TextField was pushed down down closer to the `Cancel` and `Continue` buttons. when the FormControl component was added
- autoFocus on TextField no longer works when the modal is open.
- The who component had to be wrapped with `<form> </form>` tags for it to work

I do agree the form approach is much simpler, and is the way to go. 

Am i using the form incorrectly in this case?

Reply:
I have added an example of a working form (though not very sexy) to your branch. If possible, please try and implement it in this branch the same way. :)